argmin/src/solver/powell/mod.rs

```diff
@@ -0,0 +1,97 @@
+use crate::core::{
+    ArgminFloat, CostFunction, DeserializeOwnedAlias, Executor, IterState, LineSearch,
+    OptimizationResult, SerializeAlias, Solver, State,
+};
+use argmin_math::{ArgminAdd, ArgminDot, ArgminSub, ArgminZeroLike};
+#[cfg(feature = "serde1")]
+use serde::{Deserialize, Serialize};
+use std::mem;
+
+#[derive(Clone)]
+#[cfg_attr(feature = "serde1", derive(Serialize, Deserialize))]
+pub struct PowellLineSearch<P, L> {
+    search_vectors: Vec<P>,
+    linesearch: L,
+}
+
+impl<P, L> PowellLineSearch<P, L> {
+    pub fn new(initial_search_vectors: Vec<P>, linesearch: L) -> Self {
+        PowellLineSearch {
+            search_vectors: initial_search_vectors,
+            linesearch,
+        }
+    }
+}
+
+impl<O, P, F, L> Solver<O, IterState<P, (), (), (), F>> for PowellLineSearch<P, L>
+where
+    O: CostFunction<Param = P, Output = F>,
```

Comment:
Since the problem is passed to a line search as well, you'll probably also require `O` to implement `Gradient`. But I'm not a 100% sure how these requirements of the LS are passed along. Maybe we just need to give it a try.

Reply:
I believe one of the advantages of Powell's method is that it doesn't need gradients. I thought that as long as the linesearch method is given a direction to search along it doesn't need gradients to be computed, i.e. the calculation of gradients should be independent from the linesearch algorith, which only searches along a provided search direction. However, I could be wrong, but I can just write tests and see if it works. I will also implement a version of this using golden section search and brent's method.